### DISCUSSION TITLE
Branch protection rules reordering

### DISCUSSION THREAD

**Discussion by wxwright** [community/community#39197] [New to GitHub]
I have quite a few branches in my repos and I want to setup different branch protection rules for them.  For example:
main
dev/*
team/*
*
\*/\*\*/\*
I discovered that if I want to add a new rule (e.g. "patch*" or "fixes/\*"), I will have to delete "*" and "\*/**/\*" branch protection rules and then add the new rule in order for the new rule to take effect.   Is there a way to reorder the branch protection rules after adding a new rule?  I looked thru the GitHub doc and did not find what I was looking for.

**Comment by queenofcorgis** [comment]
Hi there! 

If a repository has multiple protected branch rules that affect the same branches, the rules that include a specific branch name have the highest priority. If there is more than one protected branch rule that references the same specific branch name, then the branch rule created first will have higher priority.

To create an exception to an existing branch rule, you can create a new branch protection rule that is higher priority, such as a branch rule for a specific branch name.

[This doc](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule) is what I'm referencing.

**Reply by wxwright** [reply]
Hello there,
Thank you for your reply!

I already have this existing branch rule:  \*/\*\*/\*  in my current repos.  And I discovered later that I have a number of branches that require a different set of branch protection rule.  Because the number of branches that are involved is not small, therefore It will be ideal to use a pattern like this with a wild card: feature/patch\* for the new branch protection rule (instead of creating a new branch rule per specific branch).   And my existing branch rule:  \*/\*\*/\* still has the higher priority because it was created first.   For this reason, I was looking for a way to reordering the branch protection rules if possible.

**Reply by ghost** [reply]
We are running into this issue as well. There are a few possible soltuions to the problem as far as I can see:
 1. Add the ability to reorder the branch protections
 2. Have matching branch protections with the most non-wildcard characters take priority. For example imagine a branch protection `feature/*`, a branch protection `**/*`, and a branch named `feature/really-cool-redesign`. Regardless of the order in which the branch protections were added `feature/really-cool-redesign` should match `feature/*` since there are 8 overlapping characters, `feature/`, instead of only one matching non-wildcard character with `**/*`, `/`. This would be nice because it would be deterministic and would not depend on when a branch protectoin was added.
 3. Allow us to set a priority for a branch protection as an integer. 
 
 Any of these solutions would allow us to add more branch protections without needing to remove and re-add the older ones

**Comment by nisargthakkar** [comment]
+1

**Comment by Asman3316** [comment]
Notrdeng blokcen

**Comment by ghost** [comment]
@queenofcorgis, since this is a valid request, is it possible to move this from `New to GitHub` to `Pull Requests` or `Repositories`? It would be a shame for this to not be worked on because it is considered a "new user error" instead of a real bug / feature that is missing.

**Comment by DNin01** [comment]
Hey, looks like it's been a while, but GitHub has since added the successor to branch protections - rulesets. You might want to play around with them. One of the differences with rulesets is that rules that target the same branch are stacked, and they're more flexible in general, so you shouldn't have to worry about re-creating rules to get around conflicts. You can learn more about it in the [documentation](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/about-rulesets). Check out ["About rule layering"](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/about-rulesets#about-rule-layering) to learn more about what I was just talking about.
